### ISSUE TITLE
The reason it's not working

### ISSUE THREAD

**Issue by ghost** [aonez/You30fps#1]
Hey aone! Sorry, I don't read greasyfork messages so I didn't see your July 27 message until now.

You said that you wanted to turn [the script](https://greasyfork.org/en/scripts/23329-disable-youtube-60-fps-force-30-fps) into a Safari extension.

I can see why your extension is not working. You're waiting until "document is ready" and "document.body exists". By that time, YouTube has already executed its own scripts that do the format checking.

This "format support-checker" code override/injection must run as soon as possible, before YouTube's own scripts have queried the format-checking functions. So it must run at "document-start", in Tampermonkey terms. At such an early injection time, often "document.body" won't even exist.

Anyway, I just updated to version 1.5 of the script today to change the injection success check method, because YouTube (or Tampermonkey) recently [broke something strange](https://greasyfork.org/en/forum/discussion/28835/x?locale=en). That's when I saw your message at GreasyFork! ;-P

**Comment by aonez** [comment]
Thanks @SteveJobzniak! I'll try to find a way to inject it on "document-start", if Tampermonkey extension can, it must be possible.

**Comment by ghost** [comment]
Yeah, it's definitely possible. You may even be able to inject as soon as your script runs, since we hook by overriding a core javascript function inside that web tab. That function may already be available by the time your extension script runs, since the javascript engine is probably fully loaded when your script runs. Alright, good luck! :-)
